akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -270,44 +301,47 @@ private[remote] abstract class InboundCompression[T >: Null](
    */
   @tailrec final def decompressInternal(incomingTableVersion: Byte, idx: Int, attemptCounter: Int): OptionVal[T] = {
     // effectively should never loop more than once, to avoid infinite recursion blow up eagerly
-    if (attemptCounter > 2) throw new IllegalStateException(s"Unable to decompress $idx from table $incomingTableVersion. Internal state: ${state}")
+    if (attemptCounter > 2) throw new IllegalStateException(s"Unable to decompress $idx from table $incomingTableVersion. Internal tables: $tables")
 
-    val current = state
-    val oldVersion = current.oldTable.version
+    val current = tables
     val activeVersion = current.activeTable.version
 
-    if (incomingTableVersion == DecompressionTable.DisabledVersion) OptionVal.None // no compression, bail out early
-    else if (incomingTableVersion == activeVersion) {
+    if (incomingTableVersion == DecompressionTable.DisabledVersion) {
+      OptionVal.None // no compression, bail out early
+    } else if (incomingTableVersion == activeVersion) {
```

Comment:
`selectTable` will find the active also and same thing is done, so this case is not needed, afaics